superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx

```diff
@@ -540,6 +555,20 @@ const FiltersConfigForm = (
 
   const hasAdhoc = formFilter?.adhoc_filters?.length > 0;
 
+  const defaultToFirstItem = formFilter?.controlValues?.defaultToFirstItem;
+
+  useEffect(() => {
```

Comment:
I think it's independent logic and we can move it to `state` file, because this file already starts to be big

Reply:
Good idea, done